pageserver/src/repository/rocksdb.rs

```diff
@@ -801,27 +878,13 @@ impl Timeline for RocksTimeline {
     ///
     /// Memorize a full image of a page version
     ///
-    fn put_page_image(&self, tag: BufferTag, lsn: Lsn, img: Bytes) {
-        let img_len = img.len();
+    fn put_page_image(&self, tag: BufferTag, lsn: Lsn, img: Bytes) -> Result<()> {
         let key = CacheKey { tag, lsn };
         let content = CacheEntryContent::PageImage(img);
 
-        let mut val_buf = content.to_bytes();
-
-        // Zero size of page image indicates that page can be removed
-        if img_len == 0 {
-            if (val_buf[0] & UNUSED_VERSION_FLAG) != 0 {
-                // records already marked for deletion
-                return;
-            } else {
-                // delete truncated multixact page
-                val_buf[0] |= UNUSED_VERSION_FLAG;
-            }
-        }
-
         trace!("put_wal_record lsn: {}", key.lsn);
         let serialized_key = key.ser().expect("serialize CacheKey should always succeed");
-        let _res = self.db.put(serialized_key, content.to_bytes());
+        let _res = self.db.put(serialized_key, content.to_bytes())?;
```

Comment:
I don't think `_res` is needed any more?

Reply:
Fixed